tests/test-location.py

```diff
@@ -156,11 +158,34 @@ def quit_on(self, method_name):
         GLib.timeout_add_seconds(20, self.fail, 'Test timed out after ' +
                                  'waiting 20 seconds for D-Bus method call.')
 
+    def quit_on_event(self, event_uuid):
```

Comment:
nit: Consider passing 'RecordSingularEvent' as a parameter or renaming this method to something like `quit_on_singular` since there are also aggregates and sequences.